WalletWasabi/WabiSabi/Backend/Banning/CoinVerifierAuditArchiver.cs

```diff
@@ -0,0 +1,148 @@
+using NBitcoin;
+using Nito.AsyncEx;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.WabiSabi.Backend.Banning;
+
+public class CoinVerifierAuditArchiver : IAsyncDisposable
+{
+	public CoinVerifierAuditArchiver(string directoryPath)
+	{
+		IoHelpers.EnsureDirectoryExists(directoryPath);
+		BaseDirectoryPath = directoryPath;
+	}
+
+	private string BaseDirectoryPath { get; }
```

Comment:
I would use simply:

```suggestion
	private string DirectoryPath { get; }
```

and that's because when somebody uses the term "base directory", then typically it's combined with another directory and it's not the case here.

(In the codebase, we have also "workingDirectory" if one wants to follows what we have elsewhere.)